concepts/fluid-balance/adjusted-urine-output/AllUrineOuputAdjusted.sql

```diff
@@ -0,0 +1,138 @@
+-- This is urine-output adjustment corrected to 1 hour intervals, every hour on the hour.
+-- This adjustment is meant to be used for AKI calculation and other research purposes.
+-- Each ICUSTAY_ID has "T_PLUS" column that represents the hourly intervals from the beginning to the end of
+-- his urine outputevents.
+
+-- The problems with regular urine-outputs in outputevents measurements are:
+--  (1) You can have two measurements at one hour interval (e.g. 00:01, 00:59).
+--  (2) Different time interval between measurements.
+-- Since urine collection is done per unit of time from the last measurement, every value should be 
+-- corrected for the time length it represents.
+
+-- The solution is summing up:
+--  1st sample in the interval - is multiplied by the portion of time within the interval to the full length of time.
+--  1st sample of the NEXT interval  - is multiplied by the portion of time within the interval to the full length of time.
+--  Other samples in the interval - simply added.
+
+-- An hour interval without measurement within, and without measurement 2 hours before or after is defined "null".
+
+CREATE OR REPLACE TABLE `Tables.AllUrineOutputsAdjusted` 
+AS
+
+WITH uo AS (
+-- relevant urine-ouput columns
+  SELECT ICUSTAY_ID, CHARTTIME, VALUE, 
+  --   CIS method
+   If(oe.itemid < 50000, "CareVue", "MetaVision") as CIS
+  FROM `mimic-iii-256413.Tables.AllUrineOutputs` oe
+  
+    WHERE oe.ICUSTAY_ID  in (SELECT ICUSTAY_ID FROM `Tables.ICUSTAY_IDs`)
+--   ALTERNATIVE: pick specific icustay_id or iterate array of icustay_id. can limit array iteration
+--   WHERE oe.ICUSTAY_ID  in (299993)
+--   WHERE oe.ICUSTAY_ID  in (SELECT ICUSTAY_ID FROM `Tables.ICUSTAY_IDs` LIMIT 10)
```

Comment:
I usually do something like `where icustay_id < 200100` when I want a few cases for debugging purposes.